### PR TITLE
Add an endpoint for messages that are scheduled

### PR DESCRIPTION
**Pull request by jacobherrington**
Using the /messages/scheduled_broadcasts endpoint, we can retrieve a
list of upcoming scheduled campaigns and canvases.

More:
https://www.braze.com/docs/api/endpoints/messaging/schedule_messages/get_messages_scheduled/#request-parameters

### CODE REVIEW

File: spec/braze_ruby/rest/scheduled_broadcasts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  before { subject.http = http }
+
+  it "makes an http call to the scheduled broadcasts endpoint" do
+    expect_scheduled_broadcasts_http_call
+
+    subject.perform
+  end
+
+  def expect_scheduled_broadcasts_http_call
```
Comment (jonallured) [inline spec/braze_ruby/rest/scheduled_broadcasts_spec.rb:27]: I'm curious to know more about why a method is extracted here - why not have this expectation inside the body of the it block?

Comment (jacobherrington) [reply spec/braze_ruby/rest/scheduled_broadcasts_spec.rb:27]: I was honestly just copying another spec that had similar breadth.

There are quite a few opportunities to standardize patterns in this library. That might be something I look into next!

Comment (jonallured) [reply spec/braze_ruby/rest/scheduled_broadcasts_spec.rb:27]: Ohhh, I just opened the first file I found and it happened to NOT have a method extracted so yeah I see what you mean about standardizing. No problem!

**Comment by jonallured** [comment]
LGTM!! 🎉

**Pull request merged**